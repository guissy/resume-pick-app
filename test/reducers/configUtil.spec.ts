import initFile, {
  initDefaultFile,
  saveFile,
} from '../../app/features/configUtil';

describe('configUtil', () => {
  it('should handle initFile', async () => {
    expect(await initFile()).toMatchSnapshot();
  });
  it('should handle initDefaultFile', async () => {
    expect(await initDefaultFile()).toMatchSnapshot();
  });

  it('should handle saveFile', async () => {
    expect(await saveFile({})).toMatchSnapshot();
  });
});
