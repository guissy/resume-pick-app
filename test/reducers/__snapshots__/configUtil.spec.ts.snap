// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`configUtil should handle initDefaultFile 1`] = `
Object {
  "default": Array [
    Object {
      "children": Array [
        Object {
          "alias": Array [
            "react native",
            "rn",
            "weex",
          ],
          "name": "RN",
          "score": 1.5,
        },
        Object {
          "alias": Array [
            "flutter1",
            "flutter2",
            "dart",
          ],
          "name": "flutter",
          "score": 1.5,
        },
        Object {
          "alias": Array [
            "webview",
            "混生",
            "hybridapp",
            "jsbridge",
            "apicloud",
            "ionic",
            "uni-app",
            "uniapp",
            "umi-app",
            "umijs",
          ],
          "name": "hybrid",
          "score": 1.5,
        },
        Object {
          "alias": Array [
            "mobile",
            "移动端",
            "webapp",
          ],
          "name": "h5",
          "score": 0.5,
        },
        Object {
          "alias": Array [
            "reactjs",
            "redux",
            "antd",
            "ant design",
          ],
          "name": "react",
          "score": 1.5,
        },
        Object {
          "alias": Array [
            "vue2",
            "vuejs",
            "vue2",
            "element",
            "vant",
          ],
          "name": "vue",
          "score": 1.5,
        },
        Object {
          "name": "vue3",
          "score": 1.5,
        },
        Object {
          "alias": Array [
            "ng",
            "ng2",
            "angularjs",
            "angular2",
            "angular4",
            "angular5",
            "angular6",
            "angular7",
            "angular8",
          ],
          "name": "angular",
          "score": 1,
        },
        Object {
          "alias": Array [
            "微信",
            "小程序",
            "taro",
            "mpvue",
            "mpvue",
          ],
          "name": "wxapp",
          "score": 1.5,
        },
        Object {
          "alias": Array [
            "nodejs",
            "express",
            "koa",
            "koa2",
            "egg",
            "next",
            "nextjs",
            "pm2",
            "hapi",
            "nuxt",
          ],
          "name": "node",
          "score": 1.5,
        },
        Object {
          "alias": Array [
            "redux",
            "saga",
            "mobx",
            "vuex",
            "vuetify",
            "dva",
            "事件总线",
            "eventBus",
            "event bus",
            "mvvm",
          ],
          "name": "state mgr",
          "score": 1,
        },
        Object {
          "alias": Array [
            "es5",
            "es6",
            "es6+",
            "es2015",
            "es2016",
            "es2017",
            "es7",
            "es8",
            "es9",
            "ecmascript",
            "原生JS",
            "原生 JS",
          ],
          "name": "es6",
          "score": 0.8,
        },
        Object {
          "alias": Array [
            "mdn",
          ],
          "name": "w3c",
          "score": 0.5,
        },
        Object {
          "alias": Array [
            "flex",
            "grid",
            "css3",
            "rem",
            "less",
            "sass",
            "scss",
            "bem",
            "stylus",
            "cssmodules",
            "css-modules",
            "css in js",
            "css-in-js",
            "styled-components",
            "jss",
            "emotion",
            "tailwind",
            "纯CSS",
            "vw",
            "vh",
          ],
          "name": "css3",
          "score": 0.5,
        },
        Object {
          "alias": Array [
            "animate.css",
            "wow.js",
            "gsap",
            "pixi",
            "pixijs",
            "svga",
            "createjs",
            "canvas",
            "gpu",
            "webgl",
          ],
          "name": "keyframes",
          "score": 0.8,
        },
      ],
      "name": "核心项",
    },
    Object {
      "children": Array [
        Object {
          "alias": Array [
            "ts",
          ],
          "name": "typescript",
          "score": 1.5,
        },
        Object {
          "alias": Array [
            "hook",
            "hoc",
            "高阶",
          ],
          "name": "hooks",
          "score": 1.5,
        },
        Object {
          "alias": Array [
            "ReactiveX",
            "ngrx",
          ],
          "name": "rxjs",
          "score": 1.5,
        },
        Object {
          "alias": Array [
            "gql",
            "relay",
          ],
          "name": "graphql",
          "score": 1.5,
        },
        Object {
          "alias": Array [
            "mocha",
            "jasmine",
            "jest",
            "cypress",
            "puppeteer",
            "unit test",
          ],
          "calc": "max",
          "name": "test",
          "score": 1.5,
        },
        Object {
          "alias": Array [
            "github.com",
            "gitee.com",
          ],
          "name": "gayhub",
          "score": 1.5,
        },
        Object {
          "alias": Array [
            "tslint",
            "eslint",
            "jslint",
            "stylelint",
            "githook",
            "husky",
            "prettier",
            "命名规范",
            "代码规范",
            "commit规范",
            "接口规范",
          ],
          "name": "lint",
          "score": 1,
        },
        Object {
          "alias": Array [
            "pr",
            "git flow",
            "git-flow",
            "gitflow",
            "代码审查",
            "codereview",
          ],
          "name": "review",
          "score": 0.5,
        },
        Object {
          "alias": Array [
            "代码静态分析",
            "SonarQube",
          ],
          "name": "代码分析",
          "score": 0.5,
        },
        Object {
          "alias": Array [
            "个人博客",
            "博客地址",
            "csdn.net",
            "cnblogs.com",
            "juejin.im",
            "zhihu.com",
            "jianshu.com",
            "segmentfault.com",
            "oschina.net",
            "imooc.com",
            "51cto.com",
            "gitbooks.io",
          ],
          "name": "blog",
          "score": 1.5,
        },
      ],
      "name": "进阶",
    },
    Object {
      "children": Array [
        Object {
          "alias": Array [
            "\\\\b阿里巴巴",
            "\\\\b腾讯",
            "\\\\b百度",
            "\\\\b京东",
            "\\\\b美团",
            "\\\\b新浪",
            "\\\\b网易",
            "\\\\b字节跳动",
          ],
          "name": "大厂背景",
          "score": 2.5,
        },
        Object {
          "alias": Array [
            "组长",
            "负责人",
            "选型",
            "架构设计",
            "带领",
            "带队",
          ],
          "name": "leader",
          "score": 2,
        },
        Object {
          "alias": Array [
            "组织分享",
            "分配工作",
            "任务分配",
            "整体进度",
            "项目进度",
            "招聘",
            "评审",
            "排期",
            "细化",
            "多次组织",
            "技术分享",
          ],
          "name": "manager",
          "score": 2,
        },
        Object {
          "alias": Array [
            "([0-9一二三四五六七八九几十百千]+)\\\\s*(万|亿)",
          ],
          "name": "流量",
          "score": 2,
        },
        Object {
          "alias": Array [
            "(减少|降低|提升|提高|降至|降到|升至|升到)(\\\\D{0,2})\\\\d+(\\\\.\\\\d+)?(%|M|s)",
          ],
          "name": "百分点",
          "score": 2,
        },
        Object {
          "alias": Array [
            "白鹭",
            "p2",
            "pomelo",
            "cocos2d",
            "egret",
            "重力",
            "粒子",
          ],
          "name": "游戏引擎",
          "score": 1.5,
        },
        Object {
          "alias": Array [
            "sqlite3",
            "sqlite",
            "mysql",
            "postgre",
            "mariadb",
            "hive",
            "redis",
          ],
          "name": "sql",
          "score": 1,
        },
        Object {
          "alias": Array [
            "nwjs",
          ],
          "name": "electron",
          "score": 1.2,
        },
        Object {
          "alias": Array [
            "iOS原生开发",
            "objective-c",
          ],
          "name": "oc",
          "score": 1.2,
        },
        Object {
          "alias": Array [
            "hls",
            "直播流",
            "视频流",
            "jsmpeg",
          ],
          "name": "webrtc",
          "score": 1.2,
        },
        Object {
          "alias": Array [
            "比特币",
            "以太坊",
          ],
          "name": "区块链",
          "score": 1.2,
        },
        Object {
          "alias": Array [
            "mockjs",
            "rap",
            "模拟",
            "抓取",
          ],
          "name": "mock",
          "score": 0.5,
        },
        Object {
          "alias": Array [
            "hygen",
            "swagger",
          ],
          "name": "ejs",
          "score": 0.5,
        },
        Object {
          "alias": Array [
            "rebase",
            "代码分支",
          ],
          "name": "git",
          "score": 0.5,
        },
        Object {
          "alias": Array [
            "kubernetes",
            "k8s",
          ],
          "name": "docker",
          "score": 0.5,
        },
        Object {
          "alias": Array [
            "socket",
            "charles",
            "实时性",
            "实时通信",
          ],
          "name": "websocket",
          "score": 0.5,
        },
        Object {
          "alias": Array [
            "monorepo",
            "submodule",
            "无服务器",
            "severless",
            "自动化构建",
          ],
          "name": "微服务",
          "score": 0.5,
        },
        Object {
          "alias": Array [
            "密文",
            "支付",
            "md5",
            "sha1",
            "签名",
            "白名单",
            "jwt",
            "x-token",
            "crypto",
          ],
          "name": "加密",
          "score": 0.5,
        },
        Object {
          "alias": Array [
            "防抖",
            "去抖",
            "节流",
            "虚拟化",
            "debounce",
            "throttle",
            "代码分割",
            "分包",
            "错误收集",
            "埋点",
            "可读性",
            "preact",
            "类型校验",
            "类型检测",
          ],
          "name": "重构",
          "score": 0.5,
        },
        Object {
          "alias": Array [
            "async",
            "await",
            "generator",
          ],
          "name": "Promise",
          "score": 0.5,
        },
        Object {
          "alias": Array [
            "封装",
            "公共方法",
            "公共资源",
            "公共模块",
            "通用类库",
            "插件开发",
            "二次封装",
            "造轮子",
            "简化",
            "可复用",
            "自动化",
          ],
          "name": "自定义",
          "score": 0.5,
        },
        Object {
          "alias": Array [
            "毫秒",
            "攻克",
            "攻坚",
            "首屏",
            "懒加载",
            "gzip",
            "webp",
            "按需加载",
          ],
          "name": "性能调优",
          "score": 0.5,
        },
        Object {
          "alias": Array [
            "基础组件",
            "公共组件",
            "通用组件",
            "业务组件",
            "组件化",
            "模块化",
          ],
          "name": "component",
          "score": 0.5,
        },
        Object {
          "alias": Array [
            "函数式",
            "haskell",
            "f#",
          ],
          "name": "编程范式",
          "score": 0.5,
        },
        Object {
          "alias": Array [
            "极客",
            "掘金",
            "知乎",
            "思否",
            "stackoverflow",
            "mdn",
            "w3school",
            "博客园",
            "简书",
            "CSDN",
          ],
          "name": "技术博客",
          "score": 0.5,
        },
        Object {
          "name": "python",
          "score": 0.5,
        },
        Object {
          "alias": Array [
            "springcloud",
            "springboot",
            "springmvc",
            "mybatis",
            "feign",
            "gateway",
            "hystrix",
            "zookeeper",
            "dubbo",
            "jpa",
            "solr",
          ],
          "name": "java",
          "score": 0.5,
        },
        Object {
          "alias": Array [
            "nacos",
            "eureka",
            "hystrix",
            "gateway",
            "zuul",
            "feign",
          ],
          "name": "middleware",
          "score": 0.5,
        },
        Object {
          "alias": Array [
            "rocketmq",
            "rabbitmq",
          ],
          "name": "queue",
          "score": 0.5,
        },
        Object {
          "name": "go",
          "score": 0.5,
        },
        Object {
          "name": "rust",
          "score": 0.5,
        },
        Object {
          "name": "cpp",
          "score": 0.5,
        },
        Object {
          "name": "swift",
          "score": 0.5,
        },
        Object {
          "name": "kotlin",
          "score": 0.5,
        },
        Object {
          "name": "scala",
          "score": 0.5,
        },
        Object {
          "alias": Array [
            "sequelize",
            "typeorm",
            "orm",
            "nosql",
          ],
          "name": "mongodb",
          "score": 0.5,
        },
        Object {
          "alias": Array [
            "shell",
            "nginx",
            "linux",
            "cicd",
            "构建速度",
            "verdaccio",
            "私有npm",
            "jenkins",
          ],
          "name": "devops",
          "score": 0.5,
        },
        Object {
          "name": "vim",
          "score": 0.5,
        },
        Object {
          "name": "markdown",
          "score": 0.5,
        },
        Object {
          "name": "ssr",
          "score": 0.5,
        },
        Object {
          "alias": Array [
            "stroybook",
          ],
          "name": "mdx",
          "score": 0.5,
        },
        Object {
          "alias": Array [
            "可视化",
            "大屏",
            "echarts",
            "d3",
            "antv",
          ],
          "name": "bi",
          "score": 0.5,
        },
        Object {
          "name": "three.js",
          "score": 0.5,
        },
        Object {
          "alias": Array [
            "winston",
            "chalk",
          ],
          "name": "log4js",
          "score": 0.5,
        },
        Object {
          "alias": Array [
            "charles",
            "fiddler",
          ],
          "name": "postman",
          "score": 0.5,
        },
        Object {
          "alias": Array [
            "力扣",
            "leet code",
          ],
          "name": "leetcode",
          "score": 0.5,
        },
        Object {
          "name": "deno",
          "score": 0.5,
        },
        Object {
          "name": "seo",
          "score": 0.5,
        },
        Object {
          "alias": Array [
            "web assembly",
          ],
          "name": "wasm",
          "score": 0.5,
        },
        Object {
          "name": "逆向",
          "score": 0.5,
        },
        Object {
          "alias": Array [
            "项目链接",
            "作品链接",
            "(由|从)(0|零)到(1|一)",
          ],
          "name": "代表作",
          "score": 0.5,
        },
        Object {
          "alias": Array [
            "数据结构与算法",
          ],
          "name": "机器学习",
          "score": 0.5,
        },
        Object {
          "alias": Array [
            "coder",
            "coding",
          ],
          "name": "coding",
          "score": 0.1,
        },
        Object {
          "alias": Array [
            "flash",
            "dreamweaver",
            "div+css",
            "v-for",
            "v-if",
          ],
          "name": "iframe",
          "score": -1.5,
        },
        Object {
          "name": "mvc",
          "score": -0.5,
        },
        Object {
          "alias": Array [
            "ajax",
            "jsonp",
            "ie6",
            "ie7",
            "ie8",
          ],
          "name": "ajax",
          "score": -1,
        },
        Object {
          "alias": Array [
            "jq",
            "zepto",
            "knockout",
            "knockoutjs",
            "extjs",
            "backbone",
            "dojo",
            "mootools",
            "emberjs",
            "yui3",
            "prototype.js",
          ],
          "name": "jquery",
          "score": -0.5,
        },
        Object {
          "alias": Array [
            "requirejs",
            "gulp",
            "grunt",
            "yeo",
            "yeoman",
            "seajs",
            "phonegap",
            "jasmine",
          ],
          "name": "svn",
          "score": -1,
        },
        Object {
          "alias": Array [
            "为人和蔼",
            "吃苦耐劳",
            "上进",
            "了解",
          ],
          "name": "积极乐观",
          "score": -0.5,
        },
        Object {
          "name": "熟悉",
          "score": -0.3,
        },
      ],
      "name": "其他",
    },
  ],
}
`;

exports[`configUtil should handle initFile 1`] = `undefined`;

exports[`configUtil should handle saveFile 1`] = `"{}"`;
